lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1634,11 +1465,11 @@ class BuildWhileGroups : public FuncOpPartialLoweringPattern {
                            PatternRewriter &rewriter) const override {
     LogicalResult res = success();
     funcOp.walk([&](Operation *op) {
-      // Only work on ops that support the WhileOpInterface.
-      if (!WhileOpInterface::isSupported(op))
+      // Only work on ops that support the ScfWhileOp.
+      if (!isa<scf::WhileOp>(op))
         return WalkResult::advance();
 
-      WhileOpInterface whileOp(op);
+      ScfWhileOp whileOp(dyn_cast<scf::WhileOp>(op));
```

Comment:
nit: you can use `cast` here, or do the `dyn_cast` above, and check if that returned null.